src/main/java/com/patsi/controller/SocialMediaController.java

```diff
@@ -0,0 +1,42 @@
+package com.patsi.controller;
+
+
+import com.patsi.bean.Recipe;
+import com.patsi.bean.SocialMediaUser;
+import com.patsi.database.repository.SocialMediaRepository;
+import com.patsi.service.SocialMediaService;
+import jakarta.validation.Valid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@Validated
+@RequestMapping("/socialMedia")
+@CrossOrigin
+public class SocialMediaController {
+    Logger log = LoggerFactory.getLogger(RecipeController.class);
+
+    @Autowired
+    private SocialMediaService socialMediaService;
+
+    @PostMapping
+    public void registerRecipe(@RequestBody SocialMediaUser user) {
+        log.info("Inside Controller Edit user Profile");
+        socialMediaService.editSocialMediaAccount(user);
+    }
+
+    @GetMapping
+    public SocialMediaUser getUserById(@RequestParam SocialMediaUser user) {
```

Comment:
Why you need a all value of the SocialMediaUser in a get request?.
Think user id should be sufficient?

Reply:
The explore page might need to get All Users?

Reply:
I mean user id should be sufficient enough for you to query the user. You don't need all data of the SocialMediaUser.